napari/_qt/dialogs/activity_dialog.py

```diff
@@ -0,0 +1,217 @@
+from pathlib import Path
+
+from qtpy.QtCore import QPoint, QSize, Qt
+from qtpy.QtGui import QMovie
+from qtpy.QtWidgets import (
+    QDialog,
+    QFrame,
+    QGraphicsOpacityEffect,
+    QHBoxLayout,
+    QLabel,
+    QScrollArea,
+    QSizePolicy,
+    QToolButton,
+    QVBoxLayout,
+    QWidget,
+)
+
+import napari.resources
+
+from ...utils.translations import trans
+from ..widgets.qt_progress_bar import ProgressBar, ProgressBarGroup
+
+
+class ActivityToggleItem(QWidget):
+    """Toggle button for Activity Dialog with progress indicator displayed
+    when there are progress bars present.
+    """
+
+    def __init__(self, parent=None) -> None:
+        super().__init__(parent=parent)
+        self.setLayout(QHBoxLayout())
+
+        self._activityBtn = QToolButton()
+        self._activityBtn.setObjectName("QtActivityButton")
+        self._activityBtn.setToolButtonStyle(Qt.ToolButtonTextBesideIcon)
+        self._activityBtn.setArrowType(Qt.UpArrow)
+        self._activityBtn.setIconSize(QSize(11, 11))
+        self._activityBtn.setText(trans._('activity'))
+        self._activityBtn.setCheckable(True)
+
+        self._inProgressIndicator = QLabel(trans._("in progress..."), self)
+        sp = self._inProgressIndicator.sizePolicy()
+        sp.setRetainSizeWhenHidden(True)
+        self._inProgressIndicator.setSizePolicy(sp)
+        load_gif = str(Path(napari.resources.__file__).parent / "loading.gif")
+        mov = QMovie(load_gif)
+        mov.setScaledSize(QSize(18, 18))
+        self._inProgressIndicator.setMovie(mov)
```

Comment:
Is there a way to have this always run in its own thread? It's jarringly slow in the segmentation example, gorgeous in the threading example.

Reply:
By the way, yes, my suggestion is that we have an indicator that is maybe grayed out when there is no activity, and blue/animated when there is, and that is the whole button. Having "activity" permanently there... I find it pretty ugly. Am I the only one who thinks that?

Reply:
maybe this is something for a followup PR?